dvc/logger.py

```diff
@@ -58,97 +52,57 @@ class ColorFormatter(logging.Formatter):
     }
 
     def format(self, record):
+        info = record.msg
+
         if record.levelname == "INFO":
-            return record.msg
-
-        if record.levelname == "ERROR" or record.levelname == "CRITICAL":
-            exception, stack_trace = self._parse_exc(record.exc_info)
-
-            return (
-                "{color}{levelname}{nc}: {description}" "{stack_trace}\n"
-            ).format(
-                color=self.color_code.get(record.levelname, ""),
-                nc=colorama.Fore.RESET,
-                levelname=record.levelname,
-                description=self._description(record.msg, exception),
-                msg=record.msg,
-                stack_trace=stack_trace,
+            return info
+
+        if record.exc_info:
+            _, exception, _ = record.exc_info
+
+            info = "{message}{separator}{exception}".format(
+                message=record.msg or "",
+                separator=" - " if record.msg and exception.args else "",
+                exception=": ".join(self._causes(exception)),
             )
 
-        return "{color}{levelname}{nc}: {msg}".format(
-            color=self.color_code.get(record.levelname, ""),
-            nc=colorama.Fore.RESET,
+            if self._current_level() == logging.DEBUG:
+                trace = "".join(traceback.format_exception(*record.exc_info))
```

Comment:
will this traceback contain all causes tracebacks too as usual in py3? So with "during handling the above exception, another exception occurred"? Just making sure 🙂

Reply:
@efiop , it will. the `test_nested_exceptions` tests this :)

Reply:
Should test for implicit chaining, i.e. context too.